tests/ecbuild_install_dependency_with_soversion/project/CMakeLists.txt

```diff
@@ -0,0 +1,16 @@
+cmake_minimum_required( VERSION 3.18 FATAL_ERROR )
+find_package(ecbuild 3.12 REQUIRED HINTS ${CMAKE_SOURCE_DIR})
+project(project VERSION 1.0 LANGUAGES CXX)
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+set(CMAKE_BUILD_WITH_INSTALL_RPATH ON)
+set(CMAKE_INSTALL_RPATH_USE_LINK_PATH OFF)
+ecbuild_find_package(NAME foo VERSION 1.33 REQUIRED)
+
+ecbuild_add_executable(
+    TARGET project
+    SOURCES main.cc
+    LIBS foo::foo
+)
+ecbuild_install_dependency_with_soversion(foo::foo)
```

Comment:
I have tried to change the type of foo::foo target to be compiled as static library, and the result is an error:
```
 CMake Error at ***/ecbuild/cmake/ecbuild_install_dependency_with_soversion.cmake:62 (message):
 No shared libraries found to install for target foo::foo
 Call Stack (most recent call first):
 CMakeLists.txt:16 (ecbuild_install_dependency_with_soversion)
```
I don't think it should be the responsibility of downstream packages to figure out if a dependency library was shared or not to conditionally call this function.

Reply:
I disagree here, it is absolutely the responsibility of the project including its dependencies to know if it is a shared or static library. Distributing static libraries for example makes only sense if you install a static library, when you are installing an executable or shared library the content of the dependency is already inside those and there is nothing to distribute here.

Reply:
OK if `ecbuild_install_dependency_with_soversion` is not to be called in projects like eckit, and others.